internal/core/metadata/rest_deviceprofile_test.go

```diff
@@ -927,69 +927,91 @@ func TestAddProfileByYaml(t *testing.T) {
+		vdcMock        MockValueDescriptorClient
 		expectedStatus int
 	}{
 		{
 			"OK",
 			createDeviceProfileRequestWithFile(okBody),
 			createDBClientWithOutlines([]mockOutline{
 				{"AddDeviceProfile", []interface{}{dp}, []interface{}{TestDeviceProfileID, nil}},
+				{"GetDeviceProfileByName", []interface{}{TestDeviceProfileName}, []interface{}{contract.DeviceProfile{}, db.ErrNotFound}},
 			}),
+			MockValueDescriptorClient{},
 			http.StatusOK,
 		},
 		{
 			"Wrong content type",
 			httptest.NewRequest(http.MethodPut, AddressableTestURI, bytes.NewBuffer(okBody)),
 			nil,
+			MockValueDescriptorClient{},
 			http.StatusInternalServerError,
 		},
 		{
 			"Missing file",
 			emptyFileRequest,
 			nil,
+			MockValueDescriptorClient{},
 			http.StatusBadRequest,
 		},
 		{
 			"Empty file",
 			createDeviceProfileRequestWithFile([]byte{}),
 			nil,
+			MockValueDescriptorClient{},
 			http.StatusBadRequest,
 		},
 		{
 			"Duplicate command name",
 			createDeviceProfileRequestWithFile(dupeBody),
-			nil,
+			createDBClientWithOutlines([]mockOutline{
+				{"AddDeviceProfile", []interface{}{dp}, []interface{}{TestDeviceProfileID, nil}},
+				{"GetDeviceProfileByName", []interface{}{TestDeviceProfileName}, []interface{}{contract.DeviceProfile{}, db.ErrNotFound}},
+			}),
+			MockValueDescriptorClient{},
 			http.StatusConflict,
 		},
 		{
 			"Duplicate profile name",
 			createDeviceProfileRequestWithFile(okBody),
 			createDBClientWithOutlines([]mockOutline{
 				{"AddDeviceProfile", []interface{}{dp}, []interface{}{"", db.ErrNotUnique}},
+				{"GetDeviceProfileByName", []interface{}{TestDeviceProfileName}, []interface{}{contract.DeviceProfile{}, db.ErrNotFound}},
 			}),
+			MockValueDescriptorClient{},
 			http.StatusConflict,
 		},
 		{
 			"Empty device profile name",
 			createDeviceProfileRequestWithFile(emptyBody),
 			createDBClientWithOutlines([]mockOutline{
 				{"AddDeviceProfile", []interface{}{emptyName}, []interface{}{"", db.ErrNameEmpty}},
+				{"GetDeviceProfileByName", []interface{}{""}, []interface{}{contract.DeviceProfile{}, db.ErrNotFound}},
 			}),
+			MockValueDescriptorClient{},
 			http.StatusBadRequest,
 		},
 		{
 			"Unsuccessful database call",
 			createDeviceProfileRequestWithFile(okBody),
 			createDBClientWithOutlines([]mockOutline{
 				{"AddDeviceProfile", []interface{}{dp}, []interface{}{"", TestError}},
+				{"GetDeviceProfileByName", []interface{}{TestDeviceProfileName}, []interface{}{contract.DeviceProfile{}, db.ErrNotFound}},
 			}),
+			MockValueDescriptorClient{},
 			http.StatusInternalServerError,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			rr := httptest.NewRecorder()
-			restAddProfileByYaml(rr, tt.request, tt.dbMock, errorconcept.NewErrorHandler(logger.NewMockClient()))
+			var loggerMock = logger.NewMockClient()
+			configuration := metadataConfig.ConfigurationStruct{
+				Writable: metadataConfig.WritableInfo{EnableValueDescriptorManagement: true},
```

Comment:
Can we add a test case which covers the `ValueDescriptor` functionality that was previously missing? I think it would be useful to have tests which exercise the functionality when the `EnableValueDescriptorManagement` value is set to either `true` or `false` and we can verify the call to `device_profile.NewAddValueDescriptorExecutor` either was correctly invoked or not invoked based on the configuration.

Reply:
No problem, but I'm not good to write the test, I need some time to try that.

Reply:
I think you can use the test function [TestAddProfileByYaml](https://github.com/edgexfoundry/edgex-go/blob/b7215669fed76c3fa38af4eb79c70478837b63aa/internal/core/metadata/rest_deviceprofile_test.go#L907) as a reference. I also think that you can add a test argument to the test table which will allow you to change the `EnableValueDescriptorManagement` value for each test execution. For example,
```go
tests := []struct {
		name           string
		request        *http.Request
		dbMock         interfaces.DBClient
		vdcMock        MockValueDescriptorClient
                enableValueDescriptorManagement bool      // New test argument
		expectedStatus int
	}
```

Then in the test execution you can use that variable to set the property rather than always using `true`:
https://github.com/edgexfoundry/edgex-go/blob/b7215669fed76c3fa38af4eb79c70478837b63aa/internal/core/metadata/rest_deviceprofile_test.go#L1010

You would then need to add the newly created test argument to each of the test table entry.

Reply:
@AnthonyMBonafide Test added. Could you help review it?